rest-api-spec/src/main/resources/rest-api-spec/test/search/90_search_after.yml

```diff
@@ -320,3 +320,130 @@
   - length: { hits.hits: 1 }
   - match: { hits.hits.0._index: test }
   - match: { hits.hits.0._source.population: null }
+
+---
+"half float":
+  - skip:
+        version: " - 2.10.99"
+        reason: half_float was broken for 2.10 and earlier
+
+  - do:
+      indices.create:
+        index: test
+        body:
+          mappings:
+            properties:
+              population:
+                type: half_float
+  - do:
+      bulk:
+        refresh: true
+        index: test
+        body: |
+          {"index":{}}
+          {"population": 184.4}
+          {"index":{}}
+          {"population": 194.4}
+          {"index":{}}
+          {"population": 144.4}
+          {"index":{}}
+          {"population": 174.4}
+          {"index":{}}
+          {"population": 164.4}
+
+  - do:
+      search:
+        index: test
+        rest_total_hits_as_int: true
+        body:
+          size: 3
+          sort: [ { population: desc } ]
+  - match: { hits.total: 5 }
+  - length: { hits.hits: 3 }
+  - match: { hits.hits.0._index: test }
+  - match: { hits.hits.0._source.population: 194.4 }
+  - match: { hits.hits.1._index: test }
+  - match: { hits.hits.1._source.population: 184.4 }
+  - match: { hits.hits.2._index: test }
+  - match: { hits.hits.2._source.population: 174.4 }
+
+  - do:
+      search:
+        index: test
+        rest_total_hits_as_int: true
+        body:
+          size: 3
+          sort: [ { population: asc } ]
+  - match: { hits.total: 5 }
+  - length: { hits.hits: 3 }
+  - match: { hits.hits.0._index: test }
+  - match: { hits.hits.0._source.population: 144.4 }
+  - match: { hits.hits.1._index: test }
+  - match: { hits.hits.1._source.population: 164.4 }
+  - match: { hits.hits.2._index: test }
+  - match: { hits.hits.2._source.population: 174.4 }
+
+  # search_after with the asc sort
+  - do:
+      search:
+        index: test
+        rest_total_hits_as_int: true
+        body:
+          size: 1
+          sort: [ { population: asc } ]
+          search_after: [ 184.375 ] # this is ruonded sort value in sort result
+  - match: { hits.total: 5 }
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0._index: test }
+  - match: { hits.hits.0._source.population: 194.4 }
+
+  # search_after with the desc sort
+  - do:
+      search:
+        index: test
+        rest_total_hits_as_int: true
+        body:
+          size: 1
+          sort: [ { population: desc } ]
+          search_after: [ 164.375 ] # this is rounded sort value in sort result
```

Comment:
This rounding is not fully explained here, can you attach the link to the documentation. If in case documentation doesn't exists, can you create a documentation request for the same?

Reply:
I have asked question to community in case such documentation exists.
https://github.com/opensearch-project/OpenSearch/issues/11128

There is a precision talk here, https://www.elastic.co/guide/en/elasticsearch/reference/current/number.html#_which_type_should_i_use
But this doesnt say sort query behaviour here like we are seeing. 

Lets take this seperately apart from this PR ?
I will follow up here https://github.com/opensearch-project/OpenSearch/issues/11128 and if we dont find document/answer, I will create doc request.

I myself was not able to find any such documentation though.